src/object.jl

```diff
@@ -372,3 +372,135 @@ function Base.read(io::PrefetchedDownloadStream, ::Type{UInt8})
     end
     return b
 end
+
+function _upload_task(io; kw...)
+    try
+        (part_n, upload_buffer) = take!(io.upload_queue)
+        # upload the part
+        parteTag, wb = uploadPart(io.store, io.url, upload_buffer, part_n, io.uploadState; io.credentials, kw...)
+        # add part eTag to our collection of eTags in the right order
+        put!(io.sync, part_n) do
+            push!(io.eTags, parteTag)
+        end
+        Base.@lock io.cond_wait begin
+            io.ntasks -= 1
+            notify(io.cond_wait)
+        end
+    catch e
+        isopen(io.upload_queue) && close(io.upload_queue, e)
```

Comment:
@Drvi Could you help me with adding a test case that triggers an error during uploading and covers these lines? I tried with uploading a part that was smaller than the minimum S3 size and got the following error, but it didn't exercise this catch block. 🤔 
```julia
HTTP.Exceptions.StatusError(400, "POST", "/jl-minio-23869/test.csv?uploadId=7e57586f-e75d-4cf6-a14e-f480ebe655cd", HTTP.Messages.Response:
  """
  HTTP/1.1 400 Bad Request
  Accept-Ranges: bytes
  Cache-Control: no-cache
  Content-Length: 364
  Content-Security-Policy: block-all-mixed-content
  Content-Type: application/xml
  Server: MinIO
  Strict-Transport-Security: max-age=31536000; includeSubDomains
  Vary: Origin, Accept-Encoding
  X-Accel-Buffering: no
  X-Amz-Request-Id: 179736FE62556FB8
  X-Content-Type-Options: nosniff
  X-Xss-Protection: 1; mode=block
  Date: Mon, 13 Nov 2023 15:04:10 GMT
 
  <?xml version="1.0" encoding="UTF-8"?>
  <Error><Code>EntityTooSmall</Code><Message>Your proposed upload is smaller than the minimum allowed object size.</Message><Key>test.csv</Key><BucketName>jl-minio-23869</BucketName><Resource>/jl-minio-23869/test.csv</Resource><RequestId>179736FE62556FB8</RequestId><HostId>7b2b4a12-8baf-4d22-bc30-e4c3f2f12bff</HostId></Error>""")
  Stacktrace:
    [1] (::HTTP.ConnectionRequest.var"#connections#4"{HTTP.ConnectionRequest.var"#connections#1#5"{HTTP.TimeoutRequest.var"#timeouts#3"{HTTP.TimeoutRequest.var"#timeouts#1#4"{HTTP.ExceptionRequest.var"#exceptions#2"
...
```

Reply:
I think you had a good idea, try writing a small file and then testing, that the channel is closed and that the exp field is populated.